analyzer/java/com/engflow/bazel/invocation/analyzer/dataproviders/FlagValueDataProvider.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2023 EngFlow Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.engflow.bazel.invocation.analyzer.dataproviders;
+
+import static com.engflow.bazel.invocation.analyzer.core.DatumSupplier.memoized;
+
+import com.engflow.bazel.invocation.analyzer.bazelprofile.BazelProfile;
+import com.engflow.bazel.invocation.analyzer.bazelprofile.BazelProfileConstants;
+import com.engflow.bazel.invocation.analyzer.core.DataProvider;
+import com.engflow.bazel.invocation.analyzer.core.DatumSupplierSpecification;
+import com.engflow.bazel.invocation.analyzer.core.InvalidProfileException;
+import com.engflow.bazel.invocation.analyzer.core.MissingInputException;
+import com.engflow.bazel.invocation.analyzer.core.NullDatumException;
+import java.util.List;
+
+/**
+ * A {@link DataProvider} that supplies data on whether specific Bazel flags were used when the
+ * Bazel profile was generated.
+ */
+public class FlagValueDataProvider extends DataProvider {
```

Comment:
```suggestion
public final class FlagValueDataProvider extends DataProvider {
```

Reply:
I can see a use case for `DataProvider`s being extended. For example, suppose we have other inputs available that let us determine more accurately the returned `Datum`s. Extending a provider would allow reusing the code presented, but extending it by other analysis. This applies to all `DataProvider`s, not just this one in particular.

(A simple example is our internal BIA, which has both the Bazel profile, as well as our EngFlow profile.)